tests/link/jax/test_random.py

```diff
@@ -24,7 +24,40 @@ def random_function(*args, **kwargs):
     with pytest.warns(
         UserWarning, match=r"The RandomType SharedVariables \[.+\] will not be used"
     ):
-        return function(*args, **kwargs)
+        return pytensor.function(*args, **kwargs)
+
+
+@pytest.mark.parametrize("random_type", ("generator", "state"))
+def test_rng_io(random_type):
+    """Test explicit (non-shared) input and output RNG types in JAX."""
+    if random_type == "generator":
+        rng = random_generator_type("rng")
+        np_rng = np.random.default_rng(0)
+    else:
+        rng = random_state_type("rng")
+        np_rng = np.random.RandomState(0)
+    jx_rng = jax.random.PRNGKey(0)
+
+    next_rng, x = aer.normal(rng=rng).owner.outputs
+
+    with pytest.warns(
+        UserWarning,
+        match="RandomTypes are implicitly converted to random PRNGKey arrays",
```

Comment:
Tangental to this PR, but what is the purpose of this PR? As a user I see it every time I compile a model with RVs to JAX, but there's nothing I can do to avoid this warning yes? What danger does it alert me to?

Reply:
This tells the user that the input type of the compiled function is no longer the standard PyTensor one (such as `np.random.default_rng` or `scipy.sparse.csr`) but a JAX-specific one. Everytime the function is called the type will be coerced into that, so if performance is critical the user can pass already that one.

Because this is a very edge case (users creating PyTensor functions with explicit sparce or RNG variables) I think it's better to just have a warning than trying to provide keyword arguments or the specific types.

If adoption increases / users complaint then it may be worthwhile to revisit our approach.